client/state/lib/middleware.js

```diff
@@ -54,6 +61,38 @@ if ( desktopEnabled ) {
 	desktop = require( 'lib/desktop' ).default;
 }
 
+let processedImmediateLoginNotification = false;
+
+/**
+ * Notifies user about the fact that they were automatically logged in
+ * via a magic link.
+ *
+ * @param {function} dispatch - redux dispatch function
+ * @param {object}   action   - the dispatched action
+ * @param {function} getState - redux getState function
+ */
+const notifyAboutImmediateLoginLinkEffects = ( dispatch, action, getState ) => {
+	if ( processedImmediateLoginNotification ) {
+		return;
+	}
+	processedImmediateLoginNotification = true;
+
+	if ( ! action.query.logged_via_magic_link ) {
+		return;
+	}
+
+	const { email } = getCurrentUser( getState() );
```

Comment:
This leads to a fatal error if the user isn't logged in.  Can we do something to handle that?  (For now, since the only messages are intended for users who are logged in, I think we could just return early here if the user isn't logged in.)

This is the only issue I saw which seemed to me like it should be addressed before merging this.

Reply:
@DavidRothstein good spot! I assumed the user must be logged in to get to this URL but I will update it :)